File: newFiveAngular/src/app/model/marchandise.ts
import { Article } from "./article";
import { Ballon } from "./ballon";
import { Maillot } from "./maillot";

export class Marchandise extends Article{
 constructor(
    private _stock?: number,
    private _description?:string,
    private _ballon?:Ballon,
    private _imageUrl?:string,
    private _maillot?:Maillot)

    {
   super();
 }

 public get imageUrl(): string | undefined {
  return this._imageUrl;
}

public set imageUrl(value: string | undefined) {
  this._imageUrl = value;
}

     public get description(): string | undefined {
        return this._description;
      }
      public set description(value: string | undefined) {
        this._description = value;
      }

      public get stock(): number | undefined {
        return this._stock;
      }
      public set stock(value: number | undefined) {
        this._stock = value;
      }

      public get ballon(): Ballon | undefined {
        return this._ballon;
      }
      public set ballon(value: Ballon | undefined) {
        this._ballon = value;
      }

      public get maillot(): Maillot | undefined {
        return this._maillot;
      }
      public set maillot(value: Maillot | undefined) {
        this._maillot = value;
      }

}

File: newFive/src/main/java/formation/projetParis/newFive/entities/Marchandise.java
package formation.projetParis.newFive.entities;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.MappedSuperclass;

import com.fasterxml.jackson.annotation.JsonView;

//@MappedSuperclass
@Entity
public abstract class Marchandise extends Article {

//	@Id
//	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqMarchandise")
//    private Long id_marchandise;
	@JsonView(JsonViews.Base.class)
	private String nom;
	@JsonView(JsonViews.Base.class)
	private int stock;
	@JsonView(JsonViews.Base.class)

	private String description;
	@JsonView(JsonViews.Base.class)
	private String imageUrl;

	public Marchandise() {
	}

	public Marchandise(String nom, int stock, String description) {
		super();
		// this.id_marchandise = id_marchandise;
		this.nom = nom;
		this.stock = stock;
		this.description = description;
	}

//	public Long getId_marchandise() {
//		return id_marchandise;
//	}
//
//	public void setId_marchandise(Long id_marchandise) {
//		this.id_marchandise = id_marchandise;
//	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public int getStock() {
		return stock;
	}

	public void setStock(int stock) {
		this.stock = stock;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(String imageUrl) {
		this.imageUrl = imageUrl;
	}

//	@Override
//	public int hashCode() {
//		return Objects.hash(id_marchandise);
//	}
//
//	@Override
//	public boolean equals(Object obj) {
//		if (this == obj)
//			return true;
//		if (obj == null)
//			return false;
//		if (getClass() != obj.getClass())
//			return false;
//		Marchandise other = (Marchandise) obj;
//		return Objects.equals(id_marchandise, other.id_marchandise);
//	}

}